modal/shared_volume.py

```diff
@@ -51,7 +53,7 @@ async def write_file(self, remote_path: str, fp: BinaryIO):
         else:
             data = fp.read()
             req = api_pb2.SharedVolumePutFileRequest(shared_volume_id=self._object_id, path=remote_path, data=data)
-        await retry_transient_errors(self._client.stub.SharedVolumePutFile, req)
+        await retry_transient_errors(self._client.stub.SharedVolumePutFile, req, max_retries=20)
```

Comment:
20 retries seems like a lot? or is this because of the transfer thing in the api server?

Reply:
Yeah, I was thinking of adding similar logic to the SharedVolumePutFile to allow for larger files. Perhaps we should have a smarter retry_transient_errors that has unbounded retries in case of some specific grpc status/error that indicates something is in progress? (Something like an http 202 - accepted)

Reply:
Yeah or maybe `SharedVolumePutFile` could just return after 55s with a payload that tells the client to retry? We use a similar pattern when we wait for images to be built.